addpage.php

```diff
@@ -0,0 +1,54 @@
+<?php
+require_once('functions.php');
+$db = getDB();
+$plant_type_list = listPlantTypes(getPlantTypes($db));
+
+$error_message = '';
+$types_message = '';
+
+if (!empty($_GET)) {
```

Comment:
this only checks that the GET has something in it, not that the specific key you need exists. you need to do the empty check on `$_GET'entry_add_error']`